compiler/src/iree/compiler/Dialect/Stream/Transforms/Passes.cpp

```diff
@@ -106,9 +112,15 @@ void buildStreamTensorPassPipeline(OpPassManager &passManager,
     passManager.addPass(IREE::Stream::createAnnotateAffinitiesPass());
   }
 
+  assert((!clReplicateGlobalsPerAffinity || !clTransferGlobalsPerAffinity) &&
```

Comment:
Do not use asserts for error handling - handle errors properly by returning error messages indicating what the problem is. Asserts are disabled in most builds, and hitting an assert for most users with them enabled is a "the compiler crashed issue." Asserts should _never_ guard user input, only internal compiler invariants.